platform/security/sts/security-sts-attributequeryclaimshandler/src/main/java/org/codice/ddf/security/claims/attributequery/AttributeQueryClient.java

```diff
@@ -0,0 +1,353 @@
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.xml.sax.SAXException;
+
+import ddf.security.samlp.SamlProtocol;
+import ddf.security.samlp.SimpleSign;
+
+public class AttributeQueryClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AttributeQueryClient.class);
+
+    private static final String SAML2_PROTOCOL = "urn:oasis:names:tc:SAML:2.0:protocol";
+
+    private static final String SAML2_SUCCESS = "urn:oasis:names:tc:SAML:2.0:status:Success";
+
+    private static final String SAML2_UNKNOWN_ATTR_PROFILE = "urn:oasis:names:tc:SAML:2.0:status:UnknownAttrProfile";
+
+    private static final String SAML2_INVALID_ATTR_NAME_VALUE = "urn:oasis:names:tc:SAML:2.0:status:InvalidAttrNameOrValue";
+
+    private static final String SAML2_UNKNOWN_PRINCIPAL = "urn:oasis:names:tc:SAML:2.0:status:UnknownPrincipal";
+
+    private SimpleSign simpleSign;
+
+    private String externalAttributeStoreUrl;
+
+    private String issuer;
+
+    private String destination;
+
+    public AttributeQueryClient(SimpleSign simpleSign, String externalAttributeStoreUrl,
+            String issuer, String destination) {
+        LOGGER.debug("Initializing AttributeQueryClient.");
+
+        this.simpleSign = simpleSign;
+        this.externalAttributeStoreUrl = externalAttributeStoreUrl;
+        this.issuer = issuer;
+        this.destination = destination;
+    }
+
+    /**
+     * Creates the AttributeQuery request, the Https client to send the request, and returns
+     * the Response from the external attribute store.
+     *
+     * @return Assertion of the response.
+     */
+    public Assertion retrieveResponse(String username) {
+        return sendRequest(signSoapRequest(createRequest(username)));
+    }
+
+    private AttributeQuery createRequest(String username) {
+        LOGGER.debug("Creating SAML Protocol AttributeQuery.");
+
+        AttributeQuery attributeQuery = SamlProtocol
+                .createAttributeQuery(SamlProtocol.createIssuer(issuer),
+                        SamlProtocol.createSubject(SamlProtocol.createNameID(username)),
+                        destination);
+
+        LOGGER.debug("SAML Protocol AttributeQuery created.");
+
+        return attributeQuery;
+    }
+
+    /**
+     * Signs SOAP message of an AttributeQuery.
+     *
+     * @param attributeQuery request to be signed.
+     * @return Document of the AttributeQuery.
+     */
+    protected Document signSoapRequest(AttributeQuery attributeQuery)
+            throws AttributeQueryException {
+        try {
+            // Create and set signature for request.
+            simpleSign.signSamlObject(attributeQuery);
+        } catch (SimpleSign.SignatureException e) {
+            throw new AttributeQueryException("Error creating signature for request", e);
+        }
+
+        // Create Soap message for request.
+        Element soapElement = createSOAPPMessage(attributeQuery);
```

Comment:
I need to create the Soap message before I sign it, otherwise I'd I would only be signing the AttributeQuery within message instead of the entire message itself.